libraries/tlv-account-resolution/src/state.rs

```diff
@@ -181,13 +201,33 @@ impl ExtraAccountMetaList {
 
         let initial_accounts_len = account_infos.len() - extra_account_metas.len();
 
+        // Convert to `AccountMeta` to check resolved metas
         let provided_metas = account_infos
             .iter()
-            .map(account_meta_from_info)
+            .map(account_info_to_meta)
             .collect::<Vec<_>>();
 
         for (i, config) in extra_account_metas.iter().enumerate() {
-            let meta = config.resolve(&provided_metas, instruction_data, program_id)?;
+            let meta = {
+                // Create a list of `Ref`s so we can reference account data in the
+                // resolution step
+                let account_key_data_refs: Vec<_> = account_infos
+                    .iter()
+                    .map(|info| {
+                        let key = *info.key;
+                        let data = info.try_borrow_data()?;
+                        Ok((key, data))
+                    })
+                    .collect::<Result<Vec<(Pubkey, Ref<&mut [u8]>)>, ProgramError>>()?;
```

Comment:
nit: Ah bummer, you need to specify all the types here?

Reply:
It gets pretty hairy if you don't, but I can try to tangle with it some more!

Reply:
Ok, but don't spend more than 3 minutes on it. it's not important